duneana/CAFMaker/CAFMaker_module.cc

```diff
@@ -419,6 +594,158 @@ namespace caf {
 
 
   //------------------------------------------------------------------------------
+
+
+  void CAFMaker::FillTruthMatchingAndOverlap(art::Ptr<recob::PFParticle> const& pfp, const art::Event &evt, std::vector<TrueParticleID> &truth, std::vector<float> &truthOverlap) const{
+    auto const clockData = art::ServiceHandle<detinfo::DetectorClocksService>()->DataFor(evt);
+
+    //First getting all the hits belonging to that PFP
+    std::vector<art::Ptr<recob::Hit>> hits = dune_ana::DUNEAnaPFParticleUtils::GetHits(pfp, evt, fPandoraLabel);
+
+    TruthMatchUtils::IDToEDepositMap idToEDepositMap;
+    for (const art::Ptr<recob::Hit>& pHit : hits){
+      TruthMatchUtils::FillG4IDToEnergyDepositMap(idToEDepositMap, clockData, pHit, true);
+    }
+
+    float totalEDeposit = 0;
+    for (const auto& [id, eDeposit] : idToEDepositMap) {
+      totalEDeposit += eDeposit;
+    }
+
+    if (totalEDeposit <= 0) {
+      mf::LogWarning("CAFMaker") << "No energy deposit found for PFP with ID " << pfp->Self() << ". Skipping truth matching.";
+      return;
+    }
+
+    for (const auto& [id, eDeposit] : idToEDepositMap) {
+      if (fMCParticlesMap.count(id) == 0) {
+        mf::LogWarning("CAFMaker") << "No MCParticle found with ID " << id << " for PFP with ID " << pfp->Self() << ". Skipping.";
+        continue;
+      }
+
+      art::Ptr<simb::MCParticle> mcpart;
+      bool isPrimary;
+      int srID;
+      std::tie(mcpart, isPrimary, srID) = fMCParticlesMap.at(id);
+
+      TrueParticleID truePart;
+      truePart.type = isPrimary ? TrueParticleID::kPrimary : TrueParticleID::kSecondary;
+      truePart.ixn = 0; //Assuming a single interaction for now
+      truePart.part = srID;
+
+      truth.push_back(truePart);
+      truthOverlap.push_back(eDeposit / totalEDeposit);
```

Comment:
These two lines ultimately calculate
 - which true particle contributed the most energy
 - How much of the energy is owned by the true particle
 
Is this correct?  If so, can there be analogous calcs based on the number of owned hits?